File: programming files/remove_duplicates.py
//method remove_duplicates, takes in an array sorts it and 

def remove_duplicates(arr):
    //sorts array
    arr.sort()
    
    unique_arr = []
    removed_elements = []

    //for the elements in the array if it is the first element or
    //its not the same as the element before it, than is store it in the unique
    //array. the rest are stored in removed elements array
    for i in range(len(arr)):
        if i == 0 or arr[i] != arr[i-1]:
            unique_arr.append(arr[i])
        else:
            removed_elements.append(arr[i])
    return unique_arr, removed_elements

# Test the function
arr = [1, 2, 2, 3, 4, 4, 5]
unique_arr, removed_elements = remove_duplicates(arr)
print("Original array:", arr)
print("Unique array:", unique_arr)
print("Removed elements:", removed_elements)

File: programming files/remove_duplicates.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class remove_duplicates {

    public static List<Integer> removeDuplicates(List<Integer> arr, List<Integer> removedElements) {

        //initialize new list
        List<Integer> uniqueArr = new ArrayList<>()

        //loop through the list and check for duplicates, remove duplicates
        for (int i = 0; i < arr.size(); i++) {
            if (!uniqueArr.contains(arr.get(i)) {
                uniqueArr.add(arr.get(i));
            } else {
                removedElements.add(arr.get(i))
            }
        }
        return uniqueArr;
    }

    //main
    public static void main(String[] args) {
        // Test the function
        List<Integer> arr = Arrays.asList(1, 2, 2, 3, 4, 4, 5);
        List<Integer> removedElements = new ArrayList<>();
        List<Integer> uniqueArr = removeDuplicates(arr, removedElements);

        //display new lists after duplicate removal
        System.out.println("Original array: " + arr);
        System.out.println("Unique array: " + uniqueArr);
        System.out.println("Removed elements: " + removedElements);
    }
}
